openapi3/loader.go

```diff
@@ -16,7 +16,14 @@ import (
 	"strings"
 )
 
+// CircularReferenceError is deprecated.
+// kin-openapi will never throw this error anymore, and it's kept for compatibility reasons
+// Deprecated: CircularReferenceError is deprecated.
 var CircularReferenceError = "kin-openapi bug found: circular schema reference not handled"
+
+// CircularReferenceCounter is deprecated.
+// kin-openapi does not use this counter anymore, and it's kept for compatibility reasons
+// Deprecated: CircularReferenceCounter is deprecated.
 var CircularReferenceCounter = 3
```

Comment:
Actually just drop these. We're `v0`, meaning we advertise as potentially breaking on next release.
Simply just add a mention on the changelog section of the README, under a new `v0.126.0` section.